generator.py

```diff
@@ -84,17 +84,25 @@ async def regenerate_crate_docs(gir_exe, crate_dir, base_gir_args):
 
     # The above `gir -m doc` generates docs.md relative to the directory containing Gir.toml
     doc_path = crate_dir / doc_path
-    embed_args = ["-g", "-m", "-o", doc_path, "-d", crate_dir / "src"]
+    embed_args = ["-m", "-d", crate_dir / "src"]
 
     logs = "==> Regenerating documentation for `{}` into `{}`...\n".format(
         crate_dir, doc_path
     )
     logs += await spawn_gir(gir_exe, doc_args)
 
+    logs += "==> Stripping documentation in `{}`...\n".format(crate_dir)
+    # -n dumps stripped docs to stdout
+    _, stderr = await spawn_process("rustdoc-stripper", embed_args + ["-s", "-n"])
```

Comment:
Why are we running `rustdoc-stripper` twice btw?

Reply:
See the commit introducing this:

> generator: Strip docs before embedding new ones
>
> Otherwise documentation is repeatedly added on subsequent runs, and
> rustdoc-stripper-ignore-next won't be needed either like it was before.

We need it to first strip/prune existing docs (usually that from a previous run) before embedding new docs again.

Reply:
That's definitely not logical. Currently we only strip to go back to the original state. What you're doing is the opposite: we don't want to have the docs in the code by default. The simplest way would be to run `rustdoc-stripper` to add docs and then `git reset HEAD`.

Reply:
Correct, but that's a really annoying workflow. Contributors might also occasionally forget to add `rustdoc-stripper-ignore-next` which is now irrelevant if we never strip docs.

Also, I think you meant `git reset --hard`, `git reset` will only reset the index but not the working tree (where prior modifications from `rustdoc-stripper` reside).

Reply:
Indeed. But anyway, maybe let's use two doc commands: one to add the documentation and the other to strip it instead of doing something like this?

Reply:
This looses its value for me if I have to run/chain two commands while developing - unless we can run them simultaneously, ie. `--strip-docs --embed-docs` runs both in the right order.

Reply:
I don't see in which case you want to run both at the same time...

Reply:
Like explained in the commit message? :)

Reply:
I still don't get in which context you want to use this command outside of the doc generation in the CI.

Reply:
You actually **don't** want/need to use it during doc generation on the CI: that'll have a clean checkout (but it's a good way to enforce `// rustdoc-stripper-ignore-next`).

It is however useful **locally during development**. You see, I'm doing quite a few docs-related PRs to `gir` lately, and any change I make has to be tested of course (like this PR!). Being able to make a quick change and re-running `./generator.py --embed-docs` is a lot easier than having to `git reset --hard && ./generator.py` and potentially loosing any other changes I might have made to the gtk-rs/gstreamer-rs dir I'm working in.

Reply:
I disagree with how you do things here. I don't think it should do both in the same command: what happens if you added documentation on an item and forgot to add `// rustdoc-stripper-ignore-next`? You run the command to **generate** docs and bingo, you got your doc comment removed.

Reply:
> I disagree with how you do things here. I don't think it should do both in the same command

Hence the suggestion to put it behind two distinct flags, that can be used together?

> what happens if you added documentation on an item and forgot to add `// rustdoc-stripper-ignore-next`? You run the command to generate docs and bingo, you got your doc comment removed.

That would have happened before when running with `--features=embed-lgpl-docs`, too.

Reply:
> Hence the suggestion to put it behind two distinct flags, that can be used together?

Ah my bad, I thought reading you didn't want that. It's an acceptable solution. :+1: 

> That would have happened before when running with `--features=embed-lgpl-docs`, too.

Not with `embed-lgpl-docs`, only with `purge-lpgl-docs`. ;)

Reply:
> Not with `embed-lgpl-docs`, only with `purge-lpgl-docs`. ;)



In `gstreamer-rs` we were always calling `lgpl_docs::purge`, also with `embed-lgpl-docs`: https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/commit/3f508963e76942099abb487093a2ae30ac757ac7. It looks like gtk-rs was doing the same: https://github.com/gtk-rs/gtk-rs/pull/465/files

Reply:
Ah indeed. Well, then it's good as is. My bad.

Reply:
> Ah indeed. Well, then it's good as is. My bad.

Np, it had me confused at first. Would you still like two separate flags or is it fine to leave it as-is?

Reply:
If it's fine I'll mark this all as resolved and re-request reviews :)

Reply:
I was typing my comment and forgot to finish it haha.

So I was about to add: might be nice to add an option which simply strip the doc comments (then we can call the function from the doc generation too).

Reply:
@GuillaumeGomez Done!

It helped find one stray documentation comment in GStreamer without `// rustdoc-stripper-ignore-next`, @sdroege what do you think, shall we do a `./generator.py --strip-docs && git diff --quiet` to combat these in the CI?

Reply:
> So I was about to add: might be nice to add an option which simply strip the doc comments (then we can call the function from the doc generation too).

@GuillaumeGomez Not exactly sure what you mean, I think this was the plan outlined above. We have `--embed-docs`, `--strip-docs`, and as obvious convenience, they can be used together like `--embed-docs --strip-docs` to first do a stripping/cleaning pass (for now stuff won't be stored anywhere, so we might stick to `purge-docs` terminology as per the original `build.rs`?) followed by an embedding pass, for rapid development.

Reply:
@MarijnS95 we had that in the past with embed & then purge-lgpl docs, it's probably worth re-adding it :-)

Reply:
@bilelmoussaoui Yeah it's a good idea, currently we're loosing the docs for `fn deinit()` in `gstreamer/src/lib.rs`. However, we're _not_ loosing the docs for `gstreamer/src/clock_time.rs`. The rest of the manual documentation (only found in `gl_memory.rs` and `gl_filter.rs`, added by yours truly) is annotated `// rustdoc-stripper-ignore-next`. Curious why `clock_time.rs` is not affected...

Reply:
I mean that we should have a command to generate docs and another to strip it. And the command which adds docs should call the function which strips it first internally like you currently do.

Reply:
@GuillaumeGomez That's what this does currently, except that `--embed-docs` _does not_ strip unless also passing `--strip-docs` which is a nice tradeoff IMO. Are you okay with that?

If so this PR is ready for review, thanks!

Reply:
will do once this one gets merged :) thanks!